### PR TITLE
Horizontal Regridding MOM_IO

### PR DESCRIPTION
**Pull request by MJHarrison-GFDL**
This removes netCDF API calls contained in MOM_horizontal_regridding and replaces with calls to MOM_io routines.

### CODE REVIEW

**Review by Hallberg-NOAA — CHANGES_REQUESTED** [review]
At first glance, these changes mostly look good to me. However, any changes to the interfaces in config_src/infra/FMS1 also have to be mirrored in the corresponding interface in config_src/infra/FMS2.

**Review by Hallberg-NOAA — APPROVED** [review]

File: config_src/infra/FMS2/MOM_io_infra.F90 [hunk 1]
```diff
@@ -666,19 +674,22 @@ subroutine get_axis_data( axis, dat )
...
-    do i=1,size(axis%ax_data) ; dat(i) = axis%ax_data(i) ; enddo
-  elseif (.not.FMS2_writes) then
-    call mpp_get_axis_data( axis%AT, dat )
-  endif
+  call mpp_get_axis_data( axis%AT, dat )
 
 end subroutine get_axis_data
 
+
+!> Extracts and returns the axis data stored in an axistype.
```
Comment (Hallberg-NOAA) [inline config_src/infra/FMS2/MOM_io_infra.F90:682]: This does not seem like the right description for the subroutine `set_axis_data()`.  Rather it seems to have been copied unmodified from `get_axis_data()`.  Please correct this desciption.

File: config_src/infra/FMS2/MOM_interp_infra.F90 [hunk 2]
```diff
@@ -16,8 +17,8 @@ module MOM_interp_infra
 
 public :: horiz_interp_type, horiz_interp_init
 public :: time_interp_extern, init_extern_field, time_interp_external_init
-public :: get_external_field_info, axistype, get_axis_data
```
Comment (Hallberg-NOAA) [inline config_src/infra/FMS2/MOM_interp_infra.F90:19]: These changes in the publicly declared interfaces from the FMS2/MOM_interp_infra module do not appear to mirror changes in the FMS1/MOM_interp_infra module.  They need to be kept identical.

**Review by Hallberg-NOAA — CHANGES_REQUESTED** [review]
The additional changes appear to be moving in the right direction, but the new comments describing set_axis_data need to be corrected.  Also, there are changes to the interfaces that are declared to be public from config_src/infra/FMS2/MOM_interp_infra that do not appear to be in line with what is done for FMS1/MOM_interp_infra.  Please correct these issues before this PR is merged into dev/gfdl.

**Comment by marshallward** [comment]
I find it confusing that there is a generic `set_axis_data()` function available to both frameworks, yet only one of them can actually use the function.

Since this function appears to only be accessed inside of the framework, I feel it would be better to make this an internal (possibly private) function exclusive to FMS2 and not part of the framework API.

**Comment by MJHarrison-GFDL** [comment]
The reason set_axis_data exists in the fms2 framework is in order to copy the axis data from the mpp_axis type. In the case of FMS1, axistype is identical to mpp_axis, so it is not necessary to copy the data. I personally do not see a problem making this a null operation in one framework.  The point of adding the warning was so that in case anyone tried to call this routine when using FMS1, they would be made aware that the call can be skipped. 

I could make it private by moving to MOM_interp_infra, but it would involve moving the axistype definition and methods and create a dependency in MOM_io_infra on MOM_interp_infra, which seems backwards.

**Comment by marshallward** [comment]
I know that a different trajectory has been decided for this PR, but wanted to follow up and withdraw my objection to introducing `set_axis_data`.  I had incorrectly though that it was raising an ERROR, not a WARNING.

I have no problem with "no-op" functions like this in one framework if it is (for example) indirectly doing the work in another function.

I would have suggested that you remove the WARNING rather than print it, since stdout is already too noisy.  But given that you're going to rework this PR, it's not necessary.

**Pull request closed without merging**